### ISSUE TITLE
Page Number in Extracted Annotations

### ISSUE THREAD

**Issue by stonerl** [jlegewie/zotfile#190]
I wonder if this issue might be a limitation of poppler an pdf.js

When a pdf has a page numbering liken this: I,II,III,IV,V,1,2,3,4,5,6

The quotation is according to the page number and not he numbering in the document. So for example, page III becomes page 3 and page 6 becomes page 11.

**Comment by jlegewie** [comment]
Yes, this is a limitation of pdf.js. I created an issue for it back in 2013: https://github.com/mozilla/pdf.js/issues/3165

**Comment by stonerl** [comment]
But the poppler-tools do support page labels, don't they?

**Comment by jlegewie** [comment]
I haven't worked on the poppler-based extraction tools for years and I am not planing to do it in the future. Happy if someone wants to improve it though.

**Comment by SamyCookie** [comment]
This is a little notification that https://github.com/mozilla/pdf.js/issues/3165 has been closed, so API support for page labels has been added in pdf.js.

**Comment by davidamichelson** [comment]
Just a friendly +1 asking if the use of page labels for the extracted annotations might be included in a future version now that it is supported in pdf.js

I am not able to write this code myself but this is a feature I would love to see please. Thanks for a great tool.

**Comment by bwiernik** [comment]
@jlegewie 

I'd be happy to spend a bit of time adding support for the pageLabel property in the Extract Annotations function. Am I correct that the key code is at line 101 here:
https://github.com/jlegewie/zotfile/blob/master/chrome/content/zotfile/pdfextract/pdfjs/src/getPDFAnnotations.js

It looks like you haven't updated the fork of pdf.js since they added in support for pageLabel to the API. Do you have a list of the modifications you needed to make to pdf.js for use in Zotfile so I could merge in the current version?

**Comment by jlegewie** [comment]
That would be great! But updating pdf.js also takes some effort mainly because I assume that pdf.js underwent pretty big changes. Last time I did it (~ January 2015), the entire pdf.js code base changed so it took some time to understand where the modifications have to take place.

So if you want to look into it, I recommend starting with my pdf.js fork here: https://github.com/jlegewie/pdf.js

There are several branches. It's a pretty long time ago so I might not remember everything correctly. But here are some notes to get you started. If you end up working on it, I am happy to take a closer look and give more tips.

`current` is the master branch without any modifications. If I am correct, it should be possible to update that branch to the current version of pdf.js (or better, the most recent release and not the dev version). I think(!) that the `extract-v4.2` branch is the one that is currently used in zotfile. My commits in the branch should show the changes I made. For example, here is a bigger commit:

https://github.com/jlegewie/pdf.js/commit/9631a41d53709e355da41daed202bfefa7ff67d2

There are other commits afterwards with additional changes so I can't tell right now to what extend these are the most recent modifications. Maybe comparing `current` and `extract-v4.2` is a better way to look at the changes but I would have to check.

There also is a test frame work, which is extremely useful to show the improvements in the extraction over time. It would also be useful for updating pdf.js. It's not automatic though. Again, I can take a closer look when you end up working on this.

**Comment by sojusnik** [comment]
Did somebody make some progress on page labels yet?

**Comment by bocarlsson3rd** [comment]
I'm also wondering if someone has made some progress ;)

**Comment by mthoms27** [comment]
To have logical pagenumbers in extracted annotations would be really great ...
